### PR TITLE
feat: added sidebar scrolling

### PR DESCRIPTION
**Pull request by amritk**
Added scrolling of the sidebar stripe style

https://stripe.com/docs/api

Also feel free to remove/keep the smooth scroll over the jumpy one stripe has

### CODE REVIEW

**Review by hanspagel — APPROVED** [review]
Feels great 🌟

File: packages/api-reference/src/components/Sidebar.vue [hunk 1]
```diff
@@ -179,17 +179,48 @@ const activeSidebarItemId = computed(() => {
...
   return flattenedItems.find((item) => isVisible(item.id))?.id ?? null
 })
+
+// This offset determines how far down the sidebar the items scroll
+const SCROLL_OFFSET = -160
+const scrollerEl = ref<HTMLElement | null>(null)
+
+// Watch for the active item changing so we can scroll the sidebar
+watch(activeSidebarItemId, (activeId) => {
+  const el = document.getElementById(`active-sidebar-element-id-${activeId!}`)
```
Comment (marclave) [inline packages/api-reference/src/components/Sidebar.vue:189]: I think we should do refs here

https://vuejs.org/guide/essentials/template-refs.html#refs-inside-v-for

**Review by marclave — CHANGES_REQUESTED** [review]
left one comment to fix up, but the scrolling feels SO smooth, great work @amritk .

Im happy to merge once that's changed :) 

Specific reasoning for this is SSG would much prefer using refs

File: packages/api-reference/src/components/Sidebar.vue (hunk 1 above)
Comment (amritk) [reply packages/api-reference/src/components/Sidebar.vue:189]: > It should be noted that the ref array does not guarantee the same order as the source array.
So then how do you access the ref after? Loop through the array?

Comment (amritk) [reply packages/api-reference/src/components/Sidebar.vue]: Alright had to do some vue learning, they made some changes from vue 2. Check it out now

**Review by marclave — APPROVED** [review]
AWESOME work, lets get this in

**Pull request merged**